src/didactUriCompletionItemProviderAsciiDoc.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as vscode from 'vscode';
+import {didactProtocol, DidactUriCompletionItemProvider} from './didactUriCompletionItemProvider';
+import * as extensionFunctions from './extensionFunctions';
+
+export class DidactUriCompletionItemProviderAsciiDoc extends DidactUriCompletionItemProvider {
+
+	public provideCompletionItemsForDidactProtocol(text: string): vscode.CompletionItem[] {
+		let completions : vscode.CompletionItem[] = super.provideCompletionItemsForDidactProtocol(text);
+
+		let didactUri = this.getDidactUriFromLine(text);
+		if (didactUri) {
+			if (!didactUri.getCommandId()) {
+				// add some additional didact types here because we can't do HTML kinds of things in AsciiDoc
+				this.validateAllRequirementsCompletion("Validate All Didact Requirements", completions);
+				this.addWorkspaceFolderCompletion("Add Temporary Folder as WS Root", completions);
+			}
+		}
+
+		return completions;
+	}
+
+	public provideCompletionItemsOutsideDidactURI(lineToSearch: string) : vscode.CompletionItem[] {
+		let completions: vscode.CompletionItem[] = [];
+
+		if (lineToSearch.indexOf(didactProtocol) === -1) {
```

Comment:
would it make sense to check lineToSearch for being undefined?
`if (lineToSearch && lineToSearch.indexOf(didactProtocol) === -1) {`

Reply:
I looked into this... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf

```
So, for example: 'undefined'.indexOf() will return 0, as undefined is found at position 0 in the string undefined. 'undefine'.indexOf() however will return -1, as undefined is not found in the string undefine.
```

Reply:
So apparently it's not necessary

Reply:
neat feature.